[Pytest] Add test case ID from Test Manager

TeodoraPatrascu opened the pull request:
Changes done with this PR:

- On the Alpha tenant a Test Manager project has been created for the DU process.
Test cases for the Pytests have been added in Test Manager. Each test case has an ID.
This ID was added in the test files, for each test, for traceability between test cases and pytests.

- Add two more tests: start in background and support persistence. 
Jira https://uipath.atlassian.net/browse/ACTV-17482

- Add test to check template.json if it has the same tests configured as project.json. Jira task:
https://uipath.atlassian.net/browse/ACTV-17098

- Merge remote-tracking branch 'origin/develop' into feature/du/add_pytest

- Small changes to the annotations test. Add a dot to the end of each annotation. Add the test input data file to constants.py

- Rewrite the test_mock_folder_structure.py to support a dynamic search of the mock files. Jira task:
https://uipath.atlassian.net/browse/ACTV-16915

- Add test to check the direction of each argument. First version of test. Jira task:
https://uipath.atlassian.net/browse/ACTV-17439

- Add test for the retry scope properties. Jira task:
https://uipath.atlassian.net/browse/ACTV-17585

TeodoraPatrascu committed:
[Pytest] Add test case ID from Test Manager.
Add two more tests: start in background and support persistence.

TeodoraPatrascu committed:
[Pytest] Add test for template.json field FileInfoCollection

TeodoraPatrascu committed:
Merge remote-tracking branch 'origin/develop' into feature/du/add_pytest

# Conflicts:
#	DocumentUnderstandingProcess/Tests/TestDataGeneration/PythonTests/TestDataVariation/Nuspec_test_data.yaml

TeodoraPatrascu committed:
[Pytest] Small change to the annotations test

TeodoraPatrascu committed:
[Pytest] Rewrite the test_mock_folder_structure.py to support a dynamic search of the mock files

TeodoraPatrascu committed:
[Pytest] Write first version of the test_arguments_direction.py, add annotation test input data file path to constants, change Annotations folder name to ArgumentsVariables

TeodoraPatrascu committed:
[Pytest] Write first version of the test_retry_scope.py, make get_absolute_filenames a fixture

dragospa committed:
Fixed the fixture duplication in conftest.py (ACTV-16750)

swittenberger commented on DocumentUnderstandingProcess/Tests/conftest.py:
```diff
@@ -7,12 +7,23 @@
+    nuspec_test_data = root_test_data_variation + "Nuspec_test_data.yaml"
+    user_guide_test_data = root_test_data_variation + "UserGuide_test_data.yaml"
+    project_json_test_data = root_test_data_variation + "Project_Json_test_data.yaml"
+
```
If you managed to do it like this, please import them from the constants.py

swittenberger replied:
also, constants should be all caps.

TeodoraPatrascu committed:
[Pytest] Change some functions as fixtures and update the retry scope test.
Rewrite the arguments direction test.
Typos in the test_template_json.py

dragospa committed:
- Updated pytest_generate_tests hook
- Removed the return_test_data_location function, instead opted for constants in the conftest file

Commit:
- updated config file to fix GiveUpWritePermission.xaml constants
- moved the annotation test as it was in the incorrect folder
- updated the requirements.txt file and removed the pypdf2 dependency, replaced by pypdf

bebef1987 committed:
Update Pipeline setup

swittenberger commented:
not needed anymore. pytest2 branch was merged instead.
